### PR TITLE
Some css and js cleanup

### PR DESCRIPTION
**Pull request by linrock**
The homepage will load a fraction of a second faster since the recaptcha script isn't loaded on every page now. Otherwise, no functional change. Just a little reorganizing and removing some unused code.

- Use bootstrap 2.3.2 from cdn instead of locally
- Remove bootstrap css/js/img files
- Move base inline style to application.css
- Only load recaptcha script on signup page
- Move base inline js to application.css
- Remove unused format_sha function

Edit: more changes

- Remove repo(url) def tag
- Use %block for head tag
- Don't need a def for flash_messages
- Move diff_url to helpers.py
- Upgrade jquery to 3.4.1
- Defer js execution when possible
- Render machines/pending sections as shown/hidden based on cookie
- Remove unused meta %block

### CODE REVIEW

**Comment by tomtor** [comment]
@linrock Nice cleanups.

Could you also look at removing

https://github.com/glinscott/fishtest/blob/5b07986dab3e638292cd04d6cf95d89d9959faeb/fishtest/fishtest/templates/base.mak#L120

and the lines below?

That is some old automation stuff that I believe is no longer used and mostly integrated in the new test code now.

**Comment by ppigazzini** [comment]
DEV updated but the pages don't show up

File: fishtest/fishtest/templates/base.mak [hunk 1]
```diff
@@ -2,75 +2,47 @@
 <html>
 <head>
   <title>Stockfish Testing Framework</title>
-  <link href="/css/bootstrap.min.css" rel="stylesheet">
   <%block name="meta"/>
 
+  <link href="https://stackpath.bootstrapcdn.com/bootstrap/2.3.2/css/bootstrap.min.css" rel="stylesheet">
+  <link href="/css/application.css" rel="stylesheet">
+
   <script src="https://code.jquery.com/jquery-1.8.3.js"></script>
```
Comment (ppigazzini) [inline fishtest/fishtest/templates/base.mak:10]: (noobie writing here) if I'm not wrong we could update to
https://code.jquery.com/jquery-1.12.4.min.js

Comment (linrock) [reply fishtest/fishtest/templates/base.mak:10]: yup, upgrading to v1.12.4 is safe. v3.4.1 drops support for old browsers like IE8, which we may not care about. You could check Google Analytics to see if any testers still use old browsers. I'll switch this to v3.4.1 unless we need to support them.

**Comment by linrock** [comment]
@tomtor yup, i've removed the lines you mentioned and did some more cleanup in the meantime. i've edited the PR description with the list of new changes.

**Comment by linrock** [comment]
@ppigazzini hmm, could you try now and see if the pages show up? i made a new set of changes. also, what browser are you using?

**Comment by ppigazzini** [comment]
@linrock chrome and edge.
Internal server error
```
Apr 09 20:51:05 vps1572612706 systemd[1]: Started Fishtest Server port 6543.
Apr 09 20:51:23 vps1572612706 pserve[27656]: 2020-04-09 20:51:23,702 ERROR [waitress][waitress] Exception while serving /tests
Apr 09 20:51:23 vps1572612706 pserve[27656]: Traceback (most recent call last):
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/tweens.py", line 13, in _error_handler
Apr 09 20:51:23 vps1572612706 pserve[27656]:     response = request.invoke_exception_view(exc_info)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/view.py", line 779, in invoke_exception_view
Apr 09 20:51:23 vps1572612706 pserve[27656]:     raise HTTPNotFound
Apr 09 20:51:23 vps1572612706 pserve[27656]: pyramid.httpexceptions.HTTPNotFound: The resource could not be found.
Apr 09 20:51:23 vps1572612706 pserve[27656]: During handling of the above exception, another exception occurred:
Apr 09 20:51:23 vps1572612706 pserve[27656]: Traceback (most recent call last):
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/channel.py", line 349, in service
Apr 09 20:51:23 vps1572612706 pserve[27656]:     task.service()
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/task.py", line 169, in service
Apr 09 20:51:23 vps1572612706 pserve[27656]:     self.execute()
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/task.py", line 439, in execute
Apr 09 20:51:23 vps1572612706 pserve[27656]:     app_iter = self.channel.server.application(environ, start_response)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/proxy_headers.py", line 65, in translate_proxy_headers
Apr 09 20:51:23 vps1572612706 pserve[27656]:     return app(environ, start_response)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 270, in __call__
Apr 09 20:51:23 vps1572612706 pserve[27656]:     response = self.execution_policy(environ, self)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 279, in default_execution_policy
Apr 09 20:51:23 vps1572612706 pserve[27656]:     return request.invoke_exception_view(reraise=True)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/view.py", line 778, in invoke_exception_view
Apr 09 20:51:23 vps1572612706 pserve[27656]:     reraise_(*exc_info)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/compat.py", line 179, in reraise
Apr 09 20:51:23 vps1572612706 pserve[27656]:     raise value
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 277, in default_execution_policy
Apr 09 20:51:23 vps1572612706 pserve[27656]:     return router.invoke_request(request)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 249, in invoke_request
Apr 09 20:51:23 vps1572612706 pserve[27656]:     response = handle_request(request)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/tweens.py", line 43, in excview_tween
Apr 09 20:51:23 vps1572612706 pserve[27656]:     response = _error_handler(request, exc)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/tweens.py", line 17, in _error_handler
Apr 09 20:51:23 vps1572612706 pserve[27656]:     reraise(*exc_info)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/compat.py", line 179, in reraise
Apr 09 20:51:23 vps1572612706 pserve[27656]:     raise value
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/tweens.py", line 41, in excview_tween
Apr 09 20:51:23 vps1572612706 pserve[27656]:     response = handler(request)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 148, in handle_request
Apr 09 20:51:23 vps1572612706 pserve[27656]:     registry, request, context, context_iface, view_name
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/view.py", line 667, in _call_view
Apr 09 20:51:23 vps1572612706 pserve[27656]:     response = view_callable(context, request)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/viewderivers.py", line 460, in rendered_view
Apr 09 20:51:23 vps1572612706 pserve[27656]:     request, result, view_inst, context
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/renderers.py", line 451, in render_view
Apr 09 20:51:23 vps1572612706 pserve[27656]:     return self.render_to_response(response, system, request=request)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/renderers.py", line 474, in render_to_response
Apr 09 20:51:23 vps1572612706 pserve[27656]:     result = self.render(value, system_values, request=request)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/renderers.py", line 470, in render
Apr 09 20:51:23 vps1572612706 pserve[27656]:     result = renderer(value, system_values)
Apr 09 20:51:23 vps1572612706 pserve[27656]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid_mako/__init__.py", line 134, in __call__
Apr 09 20:51:23 vps1572612706 pserve[27656]:     raise ValueError('renderer was passed non-dictionary as value')
Apr 09 20:51:23 vps1572612706 pserve[27656]: 
```

**Comment by linrock** [comment]
@ppigazzini thanks, i found and fixed the server crash.. it should work now

**Comment by ppigazzini** [comment]
DEV updated, up and running. I start some tests, feel free to play a bit.

**Comment by ppigazzini** [comment]
@linrock buttons "Show/Hide" in home don't work.

**Comment by ppigazzini** [comment]
@linrock errata corrige: buttons work in chrome after CRTL+F5

**Comment by ppigazzini** [comment]
@linrock DEV stopped working 5 minutes after joining some workers, here below the log. Restart with `sudo systemctl fishtest@{6543..6544}` to try to debug

```
sudo journalctl -u fishtest@6543
...

Apr 09 22:31:05 vps1572612706 pserve[28246]: OSError: [Errno 24] Too many open files
Apr 09 22:31:05 vps1572612706 pserve[28246]: 2020-04-09 22:31:05,418 WARNI [waitress][MainThread] server accept() threw an exception
Apr 09 22:31:05 vps1572612706 pserve[28246]: Traceback (most recent call last):
Apr 09 22:31:05 vps1572612706 pserve[28246]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/server.py", line 311, in handle_accept
Apr 09 22:31:05 vps1572612706 pserve[28246]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/wasyncore.py", line 420, in accept
Apr 09 22:31:05 vps1572612706 pserve[28246]:   File "/usr/lib/python3.6/socket.py", line 205, in accept
Apr 09 22:31:05 vps1572612706 pserve[28246]:     fd, addr = self._accept()
A
```

**Comment by ppigazzini** [comment]
@linrock "New" test page, "SPRT bounds:" combo box starts empty. The default should be "Standard STC {-0.5, 1.5}"

**Comment by ppigazzini** [comment]
@linrock DEV updated, at the moment is working fine, continuing the tests :)

**Comment by linrock** [comment]
@ppigazzini thanks for catching those bugs. both the worker "Too many open files" and "SPRT bounds:" combo box starts empty issues should be fixed now

**Comment by ppigazzini** [comment]
@linrock after 1 hour the framework is working fine, combo box bug fixed, logs clean :)

File: fishtest/fishtest/templates/base.mak [hunk 2]
```diff
@@ -2,94 +2,79 @@
 <html>
 <head>
   <title>Stockfish Testing Framework</title>
-  <link href="/css/bootstrap.min.css" rel="stylesheet">
-  <%block name="meta"/>
 
-  <script src="https://code.jquery.com/jquery-1.8.3.js"></script>
-  <script src="/js/bootstrap.min.js"></script>
-  <script src="/js/jquery.cookie.js"></script>
+  <link href="https://stackpath.bootstrapcdn.com/bootstrap/2.3.2/css/bootstrap.min.css" rel="stylesheet">
```
Comment (ppigazzini) [inline fishtest/fishtest/templates/base.mak:6]: A couple of note (from a noobie, feel free to correct my statements):
 - the official URI for v2.3.2 is different (but there is a redirection in place), see https://www.bootstrapcdn.com/legacy/bootstrap/
 - we could use SRI hashing (the default copying the official URI)
```
<link href="https://stackpath.bootstrapcdn.com/twitter-bootstrap/2.3.2/css/bootstrap-combined.min.css" rel="stylesheet" integrity="sha384-4FeI0trTH/PCsLWrGCD1mScoFu9Jf2NdknFdFoJhXZFwsvzZ3Bo5sAh7+zL8Xgnd" crossorigin="anonymous">
```

File: fishtest/fishtest/templates/base.mak [hunk 3]
```diff
@@ -2,94 +2,79 @@
...
-      margin-right: 0;
-    }
-    th {
-      cursor: pointer;
-    }
-  </style>
+  <script src="https://code.jquery.com/jquery-3.4.1.min.js"
+          integrity="sha384-vk5WoKIaW/vJyUAd9n/wmopsmNhiy+L2Z+SBxGYnUkunIxVxAv/UtMOhba/xskxh"
+          crossorigin="anonymous"></script>
+  <script src="https://stackpath.bootstrapcdn.com/bootstrap/2.3.2/js/bootstrap.min.js"></script>
```
Comment (ppigazzini) [inline fishtest/fishtest/templates/base.mak:12]:
```
<script src="https://stackpath.bootstrapcdn.com/twitter-bootstrap/2.3.2/js/bootstrap.min.js" integrity="sha384-vOWIrgFbxIPzY09VArRHMsxned7WiY6hzIPtAIIeTFuii9y3Cr6HE6fcHXy5CFhc" crossorigin="anonymous"></script>
```

File: fishtest/fishtest/templates/tests_stats.mak [hunk 4]
```diff
@@ -25,11 +25,11 @@
 <html lang="en-us">
   <head>
     <title>Raw statistics for ${run['_id']}</title>
-    <link href="/css/bootstrap.min.css" rel="stylesheet">
+    <link href="https://stackpath.bootstrapcdn.com/bootstrap/2.3.2/css/bootstrap.min.css" rel="stylesheet">
```
Comment (ppigazzini) [inline fishtest/fishtest/templates/tests_stats.mak:28]:
```
<link href="https://stackpath.bootstrapcdn.com/twitter-bootstrap/2.3.2/css/bootstrap-combined.min.css" rel="stylesheet" integrity="sha384-4FeI0trTH/PCsLWrGCD1mScoFu9Jf2NdknFdFoJhXZFwsvzZ3Bo5sAh7+zL8Xgnd" crossorigin="anonymous">
```

File: fishtest/fishtest/templates/base.mak (hunk 2 above)
Comment (linrock) [reply fishtest/fishtest/templates/base.mak]: thanks! i've switched to using this url. i agree using the official url + SRI hashes is better

**Comment by linrock** [comment]
@ppigazzini thanks! i've updated those bootstrap CDN urls to use the official urls with SRI hashes

**Comment by ppigazzini** [comment]
@linrock Internal Server Error when checking a tests page >=1535, e.g. https://dfts-0.pigazzini.it/tests?page=1538

Here the log:
```
Apr 10 14:04:46 vps1572612706 pserve[5098]: 2020-04-10 14:04:46,704 ERROR [waitress][waitress] Exception while serving /tests
Apr 10 14:04:46 vps1572612706 pserve[5098]: Traceback (most recent call last):
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/tweens.py", line 13, in _error_handler
Apr 10 14:04:46 vps1572612706 pserve[5098]:     response = request.invoke_exception_view(exc_info)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/view.py", line 779, in invoke_exception_view
Apr 10 14:04:46 vps1572612706 pserve[5098]:     raise HTTPNotFound
Apr 10 14:04:46 vps1572612706 pserve[5098]: pyramid.httpexceptions.HTTPNotFound: The resource could not be found.
Apr 10 14:04:46 vps1572612706 pserve[5098]: During handling of the above exception, another exception occurred:
Apr 10 14:04:46 vps1572612706 pserve[5098]: Traceback (most recent call last):
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/channel.py", line 349, in service
Apr 10 14:04:46 vps1572612706 pserve[5098]:     task.service()
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/task.py", line 169, in service
Apr 10 14:04:46 vps1572612706 pserve[5098]:     self.execute()
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/task.py", line 439, in execute
Apr 10 14:04:46 vps1572612706 pserve[5098]:     app_iter = self.channel.server.application(environ, start_response)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/waitress/proxy_headers.py", line 65, in translate_proxy_headers
Apr 10 14:04:46 vps1572612706 pserve[5098]:     return app(environ, start_response)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 270, in __call__
Apr 10 14:04:46 vps1572612706 pserve[5098]:     response = self.execution_policy(environ, self)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 279, in default_execution_policy
Apr 10 14:04:46 vps1572612706 pserve[5098]:     return request.invoke_exception_view(reraise=True)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/view.py", line 778, in invoke_exception_view
Apr 10 14:04:46 vps1572612706 pserve[5098]:     reraise_(*exc_info)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/compat.py", line 179, in reraise
Apr 10 14:04:46 vps1572612706 pserve[5098]:     raise value
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 277, in default_execution_policy
Apr 10 14:04:46 vps1572612706 pserve[5098]:     return router.invoke_request(request)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 249, in invoke_request
Apr 10 14:04:46 vps1572612706 pserve[5098]:     response = handle_request(request)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/tweens.py", line 43, in excview_tween
Apr 10 14:04:46 vps1572612706 pserve[5098]:     response = _error_handler(request, exc)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/tweens.py", line 17, in _error_handler
Apr 10 14:04:46 vps1572612706 pserve[5098]:     reraise(*exc_info)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/compat.py", line 179, in reraise
Apr 10 14:04:46 vps1572612706 pserve[5098]:     raise value
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/tweens.py", line 41, in excview_tween
Apr 10 14:04:46 vps1572612706 pserve[5098]:     response = handler(request)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/router.py", line 148, in handle_request
Apr 10 14:04:46 vps1572612706 pserve[5098]:     registry, request, context, context_iface, view_name
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/view.py", line 667, in _call_view
Apr 10 14:04:46 vps1572612706 pserve[5098]:     response = view_callable(context, request)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/viewderivers.py", line 460, in rendered_view
Apr 10 14:04:46 vps1572612706 pserve[5098]:     request, result, view_inst, context
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/renderers.py", line 451, in render_view
Apr 10 14:04:46 vps1572612706 pserve[5098]:     return self.render_to_response(response, system, request=request)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/renderers.py", line 474, in render_to_response
Apr 10 14:04:46 vps1572612706 pserve[5098]:     result = self.render(value, system_values, request=request)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid/renderers.py", line 470, in render
Apr 10 14:04:46 vps1572612706 pserve[5098]:     result = renderer(value, system_values)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid_mako/__init__.py", line 156, in __call__
Apr 10 14:04:46 vps1572612706 pserve[5098]:     reraise(MakoRenderingException(errtext), None, exc_info[2])
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid_mako/compat.py", line 74, in reraise
Apr 10 14:04:46 vps1572612706 pserve[5098]:     raise value.with_traceback(tb)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid_mako/__init__.py", line 148, in __call__
Apr 10 14:04:46 vps1572612706 pserve[5098]:     result = template.render_unicode(**system)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/template.py", line 482, in render_unicode
Apr 10 14:04:46 vps1572612706 pserve[5098]:     self, self.callable_, args, data, as_unicode=True
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/runtime.py", line 883, in _render
Apr 10 14:04:46 vps1572612706 pserve[5098]:     **_kwargs_for_callable(callable_, data)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/runtime.py", line 920, in _render_context
Apr 10 14:04:46 vps1572612706 pserve[5098]:     _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/runtime.py", line 947, in _exec_template
Apr 10 14:04:46 vps1572612706 pserve[5098]:     callable_(context, *args, **kwargs)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "base_mak", line 50, in render_body
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "base_mak", line 50, in render_body
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "tests_mak", line 120, in render_body
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/runtime.py", line 795, in _include_file
Apr 10 14:04:46 vps1572612706 pserve[5098]:     callable_(ctx, **kwargs)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "run_table_mak", line 69, in render_body
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/fishtest/helpers.py", line 3, in diff_url
Apr 10 14:04:46 vps1572612706 pserve[5098]:     run['args']['tests_repo'],
Apr 10 14:04:46 vps1572612706 pserve[5098]: pyramid_mako.MakoRenderingException:
Apr 10 14:04:46 vps1572612706 pserve[5098]: Traceback (most recent call last):
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/pyramid_mako/__init__.py", line 148, in __call__
Apr 10 14:04:46 vps1572612706 pserve[5098]:     result = template.render_unicode(**system)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/template.py", line 482, in render_unicode
Apr 10 14:04:46 vps1572612706 pserve[5098]:     self, self.callable_, args, data, as_unicode=True
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/runtime.py", line 883, in _render
Apr 10 14:04:46 vps1572612706 pserve[5098]:     **_kwargs_for_callable(callable_, data)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/runtime.py", line 920, in _render_context
Apr 10 14:04:46 vps1572612706 pserve[5098]:     _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/runtime.py", line 947, in _exec_template
Apr 10 14:04:46 vps1572612706 pserve[5098]:     callable_(context, *args, **kwargs)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/fishtest/templates/base.mak", line 78, in render_body
Apr 10 14:04:46 vps1572612706 pserve[5098]:     ${self.body()}
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/fishtest/templates/tests.mak", line 101, in render_body
Apr 10 14:04:46 vps1572612706 pserve[5098]:     <%include file="run_table.mak" args="runs=runs['finished']"/>
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/env/lib/python3.6/site-packages/mako/runtime.py", line 795, in _include_file
Apr 10 14:04:46 vps1572612706 pserve[5098]:     callable_(ctx, **kwargs)
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/fishtest/templates/run_table.mak", line 70, in render_body
Apr 10 14:04:46 vps1572612706 pserve[5098]:     <td style="width:2%"><a href="${h.diff_url(run)}" target="_blank">diff</a></td>
Apr 10 14:04:46 vps1572612706 pserve[5098]:   File "/home/usr00/fishtest/fishtest/fishtest/helpers.py", line 3, in diff_url
Apr 10 14:04:46 vps1572612706 pserve[5098]:     run['args']['tests_repo'],
Apr 10 14:04:46 vps1572612706 pserve[5098]: KeyError: 'tests_repo'
```

I'm using the DB from PROD (usual indexes)
```
$ ${VENV}/bin/python x/show_indexes.py

Current Indexes:
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.actions', 'v': 2},
 'username_1': {'key': [('username', 1)], 'ns': 'fishtest_new.actions', 'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.clop', 'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.deltas', 'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.flag_cache', 'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.old_runs', 'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.old_user_cache', 'v': 2}}
{}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.regression_data', 'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.runs', 'v': 2},
 'args.username_1': {'background': True,
                     'key': [('args.username', 1)],
                     'ns': 'fishtest_new.runs',
                     'v': 2},
 'finished_1_last_updated_-1_start_time_-1': {'background': True,
                                              'key': [('finished', 1),
                                                      ('last_updated', -1),
                                                      ('start_time', -1)],
                                              'ns': 'fishtest_new.runs',
                                              'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.top_month', 'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.user_cache', 'v': 2}}
{'_id_': {'key': [('_id', 1)], 'ns': 'fishtest_new.users', 'v': 2},
 'username_1': {'key': [('username', 1)],
                'ns': 'fishtest_new.users',
                'unique': True,
                'v': 2}}
```

**Comment by linrock** [comment]
@ppigazzini good catch, i pushed a fix for it. I believe the problem was that `tests_repo` key doesn't exist for the oldest tests and should still default to `https://github.com/mcostalba/FishCooking`

**Comment by ppigazzini** [comment]
PROD updated, magnificent job @linrock :)

**Pull request merged**